src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,220 @@
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(OperatorIndexerCompletionProvider), LanguageNames.CSharp), Shared]
+    [ExtensionOrder(After = nameof(SymbolCompletionProvider))]
+    internal class OperatorIndexerCompletionProvider : LSPCompletionProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public OperatorIndexerCompletionProvider()
+        {
+        }
+
+        private const string MinimalTypeNamePropertyName = "MinimalTypeName";
+        private const string CompletionHandlerPropertyName = "CompletionHandler";
+        private const string CompletionHandlerConversion = "Conversion";
+        private const string CompletionHandlerIndexer = "Indexer";
+
+        internal override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static ImmutableDictionary<string, string> CreateCompletionHandlerProperty(string operation, params (string, string)[] otherKVPs)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder.Add(CompletionHandlerPropertyName, operation);
+            foreach (var (key, value) in otherKVPs)
+            {
+                builder.Add(key, value);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            if (!(token.IsKind(SyntaxKind.DotToken) | token.IsKind(SyntaxKind.IdentifierToken)))
```

Comment:
instead of looking for identifier, use the extension GetPreviousTokenIfTouchingWord.    then you can check if that token is a dot.

Reply:
Done in 517fd0ae (one test case changed: `c.fl $$`: Before suggestion, now no suggestion.)